### PR TITLE
Add mocking location functionality.

### PR DESCRIPTION
**Pull request by ghost**
Add `mockLocation` method for generation random locations.

### CODE REVIEW

File: lib/src/mock_location.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+class Location {
+  double lat;
+  double lon;
+  Location(this.lat, this.lon);
+}
+
+/// Generate random latitude longitude in `radius` from `centerLat` to `centerLon`, inclusive.
+/// If any argument is null, random location will be returned.
+Location mockLocation([double centerLat, double centerLon, int radius]) {
```
Comment (dinkopehar) [inline lib/src/mock_location.dart]: `mockData` uses builtin types. I think better approach would be to use `Map<String, double>` return type here. I try to keep `mockData` as simple as possible with readable code base.

Comment (dinkopehar) [inline lib/src/mock_location.dart]: `centerLat` and `centerLon` can be renamed to `latitude` and `longitude` ? I don't see reason to include `center` as prefix ?

File: lib/src/mock_location.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  if (centerLat == null || centerLon == null || radius == null) {
+    return Location(random.nextDouble() * 90, random.nextDouble() * 180);
```
Comment (dinkopehar) [inline lib/src/mock_location.dart]: I see you used Decimal Degrees (DD). It states at https://en.wikipedia.org/wiki/Decimal_degrees that values can go from -90 to 90 for latitude and -180 to 180 for longitude. Is this right ?

File: test/mock_data_test.dart [hunk 3]
```diff
@@ -152,4 +154,18 @@ void main() {
 
   });
 
+  group('mockLocation tests', () {
+
+    test('Test mockLocation only', () {
+      final location = mockLocation();
+      expect(location.lat, lessThanOrEqualTo(90.0));
+      expect(location.lat, lessThanOrEqualTo(180.0));
```
Comment (dinkopehar) [inline test/mock_data_test.dart]: I think you meant `location.lon` here ?

**Review by dinkopehar — CHANGES_REQUESTED** [review]
Hi and welcome to GitHub. I see this is your first time contributing, as stated by Github in your comment above.

`mockData` uses builtin data types. I think in that way, users can really benefit from using types such as Strings, Integers, Maps and such. These types are documented and native. Your idea to add `mockLocation` is great ! I wrote you comments on which lines can you modify to make it more simple.
We can work this out so I can merge your pull request and publish new version of `mockData` with you as contributor :tada:

File: lib/src/mock_location.dart (hunk 2 above)
Comment (ghost) [reply lib/src/mock_location.dart]: You right. I missed negative ranges.

Comment (dinkopehar) [reply lib/src/mock_location.dart]: Yeah, I see. Try to fix this and other review comments so I can successfully merge and resolve this pull request :smile: :rocket:

File: lib/src/mock_location.dart (hunk 1 above)
Comment (ghost) [reply lib/src/mock_location.dart]: Done. Now `mockLocation` returns `Map` with keys `lat` and `lon`.

File: test/mock_data_test.dart (hunk 3 above)
Comment (ghost) [reply test/mock_data_test.dart]: Exactly! Just typo.

**Comment by dinkopehar** [comment]
@kifio One more thing before I merge ... Can you please tell me what `distance` method is used for ? Other than that, this looks good.

**Comment by ghost** [comment]
> @kifio One more thing before I merge ... Can you please tell me what `distance` method is used for ? Other than that, this looks good.

@dinko-pehar Actually, this method uses only for tests. So, maybe it would be better to move this method to unit tests.

**Comment by dinkopehar** [comment]
It's OK for now. I can take it from here. Will modify it a little bit and add you to contributors list. You did great on your first pull request. Congratulations :tada: :tada: :tada:

**Comment by ghost** [comment]
> It's OK for now. I can take it from here. Will modify it a little bit and add you to contributors list. You did great on your first pull request. Congratulations 🎉 🎉 🎉

Thank you :)

**Review by dinkopehar — APPROVED** [review]
Great ! This is better now. You did successfully made changes.

**Pull request merged**